test/cljs/rems/test_table.cljs

```diff
@@ -0,0 +1,81 @@
+(ns rems.test-table
+  (:require [cljs.test :refer-macros [deftest is testing]]
+            [rems.table :refer [apply-row-defaults parse-search-terms]]))
+
+(deftest test-apply-row-defaults
+  (testing "all custom"
+    (is (= {:key 123
+            :foo {:sort-value "foo1"
+                  :display-value "foo2"
+                  :filter-value "foo3"
+                  :td [:td "foo4"]}}
+           (apply-row-defaults {:key 123
+                                :foo {:sort-value "foo1"
+                                      :display-value "foo2"
+                                      :filter-value "foo3"
+                                      :td [:td "foo4"]}}))))
+  (testing "all defaults, string value"
+    (is (= {:key 123
+            :foo {:value "foo"
+                  :sort-value "foo"
+                  :display-value "foo"
+                  :filter-value "foo"
+                  :td [:td {:class "foo"} "foo"]}}
+           (apply-row-defaults {:key 123
+                                :foo {:value "foo"}}))))
+  (testing "all defaults, non-string value"
+    (is (= {:key 123
+            :foo {:value 42
+                  :sort-value 42
+                  :display-value "42"
+                  :filter-value "42"
+                  :td [:td {:class "foo"} "42"]}}
+           (apply-row-defaults {:key 123
+                                :foo {:value 42}}))))
+  (testing "component only"
+    (is (= {:key 123
+            :foo {:sort-value nil
+                  :display-value ""
+                  :filter-value ""
+                  :td [:td.foo [:button "Button"]]}}
+           (apply-row-defaults {:key 123
+                                :foo {:td [:td.foo [:button "Button"]]}}))))
+  (testing ":sort-value is normalized to lowercase"
+    (is (= {:key 123
+            :foo {:value "FooBar"
+                  :sort-value "foobar"
+                  :display-value "FooBar"
+                  :filter-value ""
+                  :td [:td ""]}}
+           (apply-row-defaults {:key 123
+                                :foo {:value "FooBar"
```

Comment:
But if you have `:sort-value "FooBar"`, it won't get lower-cased. Is that intentional?

Reply:
Nope, I'll fix..

Reply:
Hmm, even though that wasn't intentional, I think it could work that way. Currently, :filter-value works analogously, and it makes sense to use the same logic for both, i.e., if someone wants to filter (or sort) explicitly by a non-lower case string, it can be done, but the default is lower-cased.

I added test cases for that for both :filter-value and :sort-value, so that the behavior is more explicit and kind of documented in the test cases.

So, in short, I'd like to keep :filter-value and :sort-value analogous in this regard, and wouldn't want to change the existing behavior of :filter-value without strong enough reasons.

Reply:
that sounds reasonable